compiler/rustc_borrowck/src/universal_regions.rs

```diff
@@ -798,6 +799,12 @@ impl<'cx, 'tcx> UniversalRegionsBuilder<'cx, 'tcx> {
                 // "output" (the type of the constant).
                 assert_eq!(self.mir_def.to_def_id(), def_id);
                 let ty = tcx.type_of(self.mir_def).instantiate_identity();
+
+                // FIXME(#129952): We probably want a more principled approach here.
+                if let Err(terr) = ty.error_reported() {
+                    self.infcx.set_tainted_by_errors(terr);
+                }
```

Comment:
I think that maybe we want to do it for all signatures, i.e. assign this match to a variable and then check whether that variable has any `error_reported`. I expect that to fix even more issues 😁 

I am otherwise quite happy with this change

Reply:
no further crashes are fixed by that change. I did rename https://github.com/rust-lang/rust/issues/129952  to reflect that the tainting now occurs in a different place.